tests/python_package_test/test_basic.py

```diff
@@ -29,6 +29,8 @@ def test(self):
             "max_bin": 255,
             "gpu_use_dp": True
         }
+        if lgb.get_device_type() == 2:
+            params["device"] = "cuda"
```

Comment:
Please make the same option as for GPU OpenCL tests instead of dozens of `params["device"] = "cuda"`.
https://github.com/microsoft/LightGBM/blob/2792923d35e7f3d392474ceffe2185c661047b2e/.ci/test.sh#L125-L127

Reply:
This is fine for CI testing but how to we set device_type to cuda for manual python unit tests?